### DISCUSSION TITLE
Card design

### DISCUSSION THREAD

**Discussion by clicksave** [keep-starknet-strange/tsubasa#11] [Ideas]
Lets use this forum to discuss card design, I.e. should we mint a set of NFTs to represent sorare cards or should we design the tool with the sorare cards in mind and place them directly in the game. 

Alex did some explores into some Anime themed soccer players. let us know if you have and feedback or recommendations. 

---

<img width="1442" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/a87f9021-bfcd-41dd-957e-0cfb2829094d">
<img width="1441" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/7693ee18-3f99-4bf2-bc85-7083aaa18f67">
<img width="1339" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/083b89f6-7154-4e86-8445-f31d422a4e99">
<img width="954" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/14df5594-57eb-4d77-acdf-5d9c1602dc1c">

**Comment by clicksave** [comment]


**Reply by ElegantShinobi** [reply]
We could also consider adding cards there are not players (like spells in hearthstone or sorceries in MTG) that can affect play as well. They would be related to football events like "Red Card", "Diving Save", "Pentalty Kick" "Offsides" etc

![image](https://github.com/keep-starknet-strange/tsubasa/assets/141261475/056c07c4-8d11-4fa3-88fe-b5fb19cc0bce)![image](https://github.com/keep-starknet-strange/tsubasa/assets/141261475/45802b4d-4268-4717-8218-601d51ec62f2)![image](https://github.com/keep-starknet-strange/tsubasa/assets/141261475/8c2823d7-fc64-4a44-97b9-8406a822f872)![image](https://github.com/keep-starknet-strange/tsubasa/assets/141261475/b6ea784e-3295-42e4-aa0f-8aa4ba998d16)![image](https://github.com/keep-starknet-strange/tsubasa/assets/141261475/5e81626f-819f-4dd7-a9c4-faea0027eb10)

**Reply by AbdelStark** [reply]
very cool idea!

**Comment by AbdelStark** [comment]
> We could also consider adding cards there are not players (like spells in hearthstone or sorceries in MTG) that can affect play as well. They would be related to football events like "Red Card", "Diving Save", "Pentalty Kick" "Offsides" etc
> 
> ![image](https://user-images.githubusercontent.com/141261475/257891079-056c07c4-8d11-4fa3-88fe-b5fb19cc0bce.png)![image](https://user-images.githubusercontent.com/141261475/257891128-45802b4d-4268-4717-8218-601d51ec62f2.png)![image](https://user-images.githubusercontent.com/141261475/257891455-8c2823d7-fc64-4a44-97b9-8406a822f872.png)![image](https://user-images.githubusercontent.com/141261475/257891477-b6ea784e-3295-42e4-aa0f-8aa4ba998d16.png)![image](https://user-images.githubusercontent.com/141261475/257891525-5e81626f-819f-4dd7-a9c4-faea0027eb10.png)



> Lets use this forum to discuss card design, I.e. should we mint a set of NFTs to represent sorare cards or should we design the tool with the sorare cards in mind and place them directly in the game.
> 
> Alex did some explores into some Anime themed soccer players. let us know if you have and feedback or recommendations.
> 
> <img alt="image" width="1442" src="https://user-images.githubusercontent.com/3332646/257883649-a87f9021-bfcd-41dd-957e-0cfb2829094d.png"> <img alt="image" width="1441" src="https://user-images.githubusercontent.com/3332646/257883727-7693ee18-3f99-4bf2-bc85-7083aaa18f67.png"> <img alt="image" width="1339" src="https://user-images.githubusercontent.com/3332646/257887829-083b89f6-7154-4e86-8445-f31d422a4e99.png"> <img alt="image" width="954" src="https://user-images.githubusercontent.com/3332646/257887921-14df5594-57eb-4d77-acdf-5d9c1602dc1c.png">

Awesome!!!

**Comment by clicksave** [comment]
OK so we have a set of components created for the cards

Here's an initial set of illustrations though we'll likely finalize the illustrations this week, we'll be rolling with 24 total (3 teams w/ 8 cards each) 

<img width="1146" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/62e8abc0-a972-4a72-a2ce-8862014f15b2">

---

Cards have attributes that need to be surfaced as well

- [ ] Dribble (top left)
- [ ] Defense (top right)
- [ ] Energy Cost (bottom center)
- [ ] Position (on hover/preview)
- [ ] Name (on preview)
- [ ] Team (background color) 
- [ ] Captain star [if captain] (top center)

<img width="700" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/89555a45-e05a-4861-966d-82290de6c432">
<img width="874" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/b30b3b5e-6719-4dea-996e-f475064e1ff7">
<img width="871" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/dbdb8e2d-2353-4c9e-94a7-ca0383c98e80">

I'll outline the states here

---

**Card (in hand)**

Pretty standard, main thing here is we need to surface the energy cost

<img width="800" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/2a90ba3a-ec09-435a-9bf0-a411da181899">

<img width="957" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/bc2e42d7-3069-454f-a239-673562967814">

https://github.com/keep-starknet-strange/tsubasa/assets/3332646/78dd217a-8aac-4e0b-a998-7269cf696c9f

[Prototype](https://www.figma.com/file/CC8hm73vxTkXSt3b5NWKvX/Tsubasa-UI-Kit?type=design&node-id=139%3A48152&mode=design&t=XTyAVzOoyY9iAsMt-1)
 
---

**Card (in play) (pending)** 

When you play a card its not immediately available. It enters a pending state before joining the board. This state lasts for a one turn.

Note: after a card is added to the board it no longer surfaces its energy cost

https://github.com/keep-starknet-strange/tsubasa/assets/3332646/65200b64-f1ed-46bb-ac03-60e60ee71600

<img width="1192" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/3bf08ed8-422d-4049-bbc6-648e58d45d24">

Hover State

<img width="526" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/1f6980a6-50ee-4ea2-840a-282d723a637a">

---

**Card (in play) (active)**

Cards in play look a lot like cards in your hand and have a similar hover effect

I have it so that captains, in addition to displaying the captain star, are a little larger than regular cards, think it makes the board more interesting, but not totally necessary

https://github.com/keep-starknet-strange/tsubasa/assets/3332646/9f86c0ce-26b0-4e9f-b77d-01d3e04015f8

---

**Card Preview**

Clicking on a card renders it in a preview window along with any stats/attributes it has

<img width="1098" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/ae9ff32e-5e24-4583-9659-9bb91a1e7135">

---

**Responsive Design**

We also have mobile iterations

<img width="1414" alt="image" src="https://github.com/keep-starknet-strange/tsubasa/assets/3332646/81d0c98e-d696-4a66-8ab5-67be9a197a96">

---

**Resources**

[Pages](https://www.figma.com/file/CC8hm73vxTkXSt3b5NWKvX/Tsubasa-UI-Kit?type=design&node-id=139%3A46259&mode=design&t=XTyAVzOoyY9iAsMt-1)
[Components](https://www.figma.com/file/CC8hm73vxTkXSt3b5NWKvX/Tsubasa-UI-Kit?type=design&node-id=74%3A17295&mode=design&t=XTyAVzOoyY9iAsMt-1)

**Reply by ElegantShinobi** [reply]
MASTERFUL WORK 👏👏

**Comment by clicksave** [comment]
Update: https://github.com/keep-starknet-strange/tsubasa/issues/5#issuecomment-1669490204

**Comment by clicksave** [comment]
**Final Illustrations** lmk if you want these in a different resolution

[Tsubasa Players_Final.zip](https://github.com/keep-starknet-strange/tsubasa/files/12361197/Tsubasa.Players_Final.zip)
